cmd/snap/cmd_snap_op.go

```diff
@@ -744,7 +744,10 @@ func (x *cmdRefresh) refreshMany(snaps []string, opts *client.SnapOptions) error
 		return showDone(x.client, chg, refreshed, "refresh", opts, x.getEscapes())
 	}
 
-	fmt.Fprintln(Stderr, i18n.G("All snaps up to date."))
+	_, timestamp := x.client.WarningsSummary()
+	if last, _ := lastWarningTimestamp(); !timestamp.After(last) {
```

Comment:
Could you please rename the timestamps names to reflect what they are? I spent some time to understand that this means that seeing the warnings using `snap warnings` will trigger the condition.

```diff
diff --git a/cmd/snap/cmd_snap_op.go b/cmd/snap/cmd_snap_op.go
index d6180b050e..c57c4dce21 100644
--- a/cmd/snap/cmd_snap_op.go
+++ b/cmd/snap/cmd_snap_op.go
@@ -744,8 +744,8 @@ func (x *cmdRefresh) refreshMany(snaps []string, opts *client.SnapOptions) error
                return showDone(x.client, chg, refreshed, "refresh", opts, x.getEscapes())
        }
 
-       _, timestamp := x.client.WarningsSummary()
-       if last, _ := lastWarningTimestamp(); !timestamp.After(last) {
+       _, lastTimestamp := x.client.WarningsSummary()
+       if lastSeenTimestamp, _ := lastWarningTimestamp(); !lastTimestamp.After(lastSeenTimestamp) {
                fmt.Fprintln(Stderr, i18n.G("All snaps up to date."))
        }
```

Reply:
OK. But it was just a copy-paste from https://github.com/snapcore/snapd/blob/31775d/cmd/snap/cmd_warnings.go#L221